x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -677,10 +676,10 @@ private LogicalPlan resolveEnrich(Enrich enrich, List<Attribute> childrenOutput)
             return enrich;
         }
 
-        private static final DataType[] GEO_TYPES = new DataType[] { GEO_POINT, GEO_SHAPE };
-        private static final DataType[] NON_GEO_TYPES = new DataType[] { KEYWORD, TEXT, IP, LONG, INTEGER, FLOAT, DOUBLE, DATETIME };
+        private static final DataTypes[] GEO_TYPES = new DataTypes[] { GEO_POINT, GEO_SHAPE };
+        private static final DataTypes[] NON_GEO_TYPES = new DataTypes[] { KEYWORD, TEXT, IP, LONG, INTEGER, FLOAT, DOUBLE, DATETIME };
```

Comment:
Maybe these lists should be pulled up into the enum?  It's also very unclear why `NON_GEO_TYPES` here is only a small subset of the remaining types.